include/i2s.h

```diff
@@ -0,0 +1,418 @@
+
+/** Data Format bit field position. */
+#define I2S_FMT_DATA_FORMAT_SHIFT           0
+/** Data Format bit field mask. */
+#define I2S_FMT_DATA_FORMAT_MASK            (0x7 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/** @brief Standard I2S Data Format.
+ *
+ * Serial data is transmitted in two’s complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the rising edge
+ * of the clock signal (SCK). The MSB is always sent one clock period after the
+ * WS changes. Left channel data are sent first indicated by WS = 0, followed
+ * by right channel data indicated by WS = 1.
+ *
+ * @verbatim
+ *      .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-.
+ * SCK -' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-
+ *     ---.                               .-------------------------------.
+ * WS     '-------------------------------'                               '-----
+ *     ---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.-
+ * SD     |   |MSB|   |...|   |LSB| x |...| x |MSB|   |...|   |LSB| x |...| x |
+ *     ---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'-
+ *            | Left channel                  | Right channel                 |
+ * @endverbatim
+ */
+#define I2S_FMT_DATA_FORMAT_I2S             (0 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/** @brief PCM Short Frame Sync Data Format.
+ *
+ * Serial data is transmitted in two’s complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the falling edge
+ * of the clock signal (SCK). The falling edge of the frame sync signal (WS)
+ * indicates the start of the PCM word. The frame sync is one clock cycle long.
+ * An arbitrary number of data words can be sent in one frame.
+ *
+ * @verbatim
+ *      .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-.
+ * SCK -' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-
+ *      .---.                                                       .---.
+ * WS  -'   '-                                                     -'   '-
+ *     -.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.-
+ * SD   |   |MSB|   |...|   |LSB|MSB|   |...|   |LSB|MSB|   |...|   |LSB|   |
+ *     -'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'-
+ *          | Word 1            | Word 2            | Word 3  |  Word n |
+ * @endverbatim
+ */
+#define I2S_FMT_DATA_FORMAT_PCM_SHORT       (1 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/** @brief PCM Long Frame Sync Data Format.
+ *
+ * Serial data is transmitted in two’s complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the falling edge
+ * of the clock signal (SCK). The rising edge of the frame sync signal (WS)
+ * indicates the start of the PCM word. The frame sync has an arbitrary length,
+ * however has to fall before the start of the next frame. An arbitrary number
```

Comment:
It refers to the frame sync signal. Should the sentence read: "however it has to fall"?